include/lsst/afw/image/Color.h

```diff
@@ -24,51 +26,56 @@ namespace image {
  */
 class Color final {
 public:
-    explicit Color(double g_r = std::numeric_limits<double>::quiet_NaN()) : _g_r(g_r) {}
+    /// Default: indeterminate color
+    Color() noexcept
+        : _color_value(std::numeric_limits<double>::quiet_NaN()),
+        _color_type(),
+        _indeterminate(true) {}
+
+    /// Fully-specified color: both a numeric value and its type string
+    Color(double color_value, std::string const & color_type) noexcept
+        : _color_value(color_value), _color_type(color_type), _indeterminate(false) {}
 
     Color(Color const &) = default;
     Color(Color &&) = default;
     Color &operator=(Color const &) = default;
     Color &operator=(Color &&) = default;
     ~Color() noexcept = default;
 
-    /// Whether the color is the special value that indicates that it is unspecified.
-    bool isIndeterminate() const noexcept { return std::isnan(_g_r); }
+    /// Whether this Color was default‑constructed (i.e. has no value/type).
+    bool isIndeterminate() const noexcept { return _indeterminate; }
+
+    /// The numeric color value; only valid if !isIndeterminate().
+    double getColorValue() const noexcept { return _color_value; }
+
+    /// The color type string (e.g. "g-r"); only valid if !isIndeterminate().
+    std::string const & getColorType() const noexcept { return _color_type; }
 
-    //@{
     /**
-     *  Equality comparison for colors
-     *
-     *  Just a placeholder like everything else, but we explicitly let indeterminate colors compare
-     *  as equal.
+     * Equality comparison for colors.
      *
-     *  In the future, we'll probably want some way of doing fuzzy comparisons on colors, but then
-     *  we'd have to define some kind of "color difference" matric, and it's not worthwhile doing
-     *  that yet.
+     * Indeterminate colors compare equal to each other; fully-specified
+     * colors compare by both value and type.
      */
-    bool operator==(Color const &other) const noexcept {
-        return (isIndeterminate() && other.isIndeterminate()) || other._g_r == _g_r;
+    bool operator==(Color const & other) const noexcept {
+        if (_indeterminate && other._indeterminate) {
+            return true;
+        }
+        if (_indeterminate != other._indeterminate) {
+            return false;
+        }
+        return _color_value == other._color_value && _color_type == other._color_type;
     }
-    bool operator!=(Color const &other) const noexcept { return !operator==(other); }
-    //@}
-
-    /// Return a hash of this object.
-    std::size_t hash_value() const noexcept { return isIndeterminate() ? 42 : std::hash<double>()(_g_r); }
```

Comment:
Why was the hash removed altogether? It should hash the type and value together.

Reply:
I don't understand the point of hashing the value here. Was maybe done when the place holder was doing nothing, but did not think it was useful at all.